frontend/src/api/k8s/__tests__/secrets.spec.ts

```diff
@@ -0,0 +1,336 @@
+    const result = await createSecret(assembleSecret('secret', data, 'aws'));
+    expect(k8sCreateResourceMock).toHaveBeenCalledWith({
+      fetchOptions: { requestInit: {} },
+      model: { apiVersion: 'v1', kind: 'Secret', plural: 'secrets' },
+      queryOptions: { queryParams: {} },
+      resource: {
+        ...assembleSecretResult,
+        stringData: { ...assembleSecretResult.stringData, type: 's3' },
+      },
+    });
+    expect(k8sCreateResourceMock).toHaveBeenCalledTimes(1);
+    expect(result).toStrictEqual(secretMock);
+  });
+
+  it('should handle errors and rethrow', async () => {
+    k8sCreateResourceMock.mockRejectedValue(new Error('error1'));
+    await expect(createSecret(assembleSecret('secret', data, 'aws'))).rejects.toThrow('error1');
+    expect(k8sCreateResourceMock).toHaveBeenCalledWith({
+      fetchOptions: { requestInit: {} },
+      model: { apiVersion: 'v1', kind: 'Secret', plural: 'secrets' },
+      queryOptions: { queryParams: {} },
+      resource: {
+        ...assembleSecretResult,
+        stringData: { ...assembleSecretResult.stringData, type: 's3' },
+      },
+    });
+    expect(k8sCreateResourceMock).toHaveBeenCalledTimes(1);
+  });
+});
+
+describe('replaceSecret', () => {
+  it('should replace the secret ', async () => {
+    const secretMock = {
+      items: [{ metadata: { name: 'item 1' } }, { metadata: { name: 'item 2' } }] as SecretKind[],
+    };
+    k8sUpdateResourceMock.mockResolvedValue(secretMock);
+    const result = await replaceSecret(assembleSecret('secret', data, 'aws'));
+    expect(k8sUpdateResourceMock).toHaveBeenCalledWith({
+      fetchOptions: { requestInit: {} },
+      model: { apiVersion: 'v1', kind: 'Secret', plural: 'secrets' },
+      queryOptions: { queryParams: {} },
+      resource: {
+        ...assembleSecretResult,
+        stringData: { ...assembleSecretResult.stringData, type: 's3' },
+      },
+    });
+    expect(k8sUpdateResourceMock).toHaveBeenCalledTimes(1);
+    expect(result).toStrictEqual(secretMock);
+  });
+
+  it('should handle errors and rethrow', async () => {
+    k8sUpdateResourceMock.mockRejectedValue(new Error('error1'));
+    await expect(replaceSecret(assembleSecret('secret', data, 'aws'))).rejects.toThrow('error1');
+    expect(k8sUpdateResourceMock).toHaveBeenCalledWith({
+      fetchOptions: { requestInit: {} },
+      model: { apiVersion: 'v1', kind: 'Secret', plural: 'secrets' },
+      queryOptions: { queryParams: {} },
+      resource: {
+        ...assembleSecretResult,
+        stringData: { ...assembleSecretResult.stringData, type: 's3' },
+      },
+    });
+    expect(k8sUpdateResourceMock).toHaveBeenCalledTimes(1);
+  });
+});
+
+describe('deleteSecret', () => {
+  it('should delete secret', async () => {
+    const secretMock = {
+      items: [{ metadata: { name: 'item 1' } }, { metadata: { name: 'item 2' } }] as SecretKind[],
+    };
+    k8sDeleteResourceMock.mockResolvedValue(secretMock);
```

Comment:
This is not the correct return value for `deleteSecret`.

Our cast to `jest.Mock` doesn't give proper typing... maybe this is something we can look at improving across all tests.